internal/provisioners/default/zz-default.provisioners.yaml

```
@@ -493,4 +493,128 @@
         type: ClusterIP
         ports:
         - port: 3306
-          targetPort: 3306
\ No newline at end of file
+          targetPort: 3306
+
+- uri: template://default-provisioners/mongo
+  type: mongo
+  init: |
+    randomDatabase: db-{{ randAlpha 8 }}
+    randomUsername: user-{{ randAlpha 8 }}
+    randomPassword: {{ randAlphaNum 16 | quote }}
+  state: |
+    service: mongo-{{ .SourceWorkload }}-{{ substr 0 8 .Guid | lower }}
+    database: {{ dig "database" .Init.randomDatabase .State | quote }}
+    username: {{ dig "username" .Init.randomUsername .State | quote }}
+    password: {{ dig "password" .Init.randomPassword .State | quote }}
+  outputs: |
+    host: {{ .State.service }}
+    port: 27017
+    name: {{ .State.database }}
+    database: {{ .State.database }}
```

Let's align this more with the implementation in `score-compose` and remove the `database` output for now and add the `connection` string output: https://github.com/score-spec/score-compose/blob/main/internal/command/default.provisioners.yaml#L502

I know it's a bit different to the other DBs above, but it's more important that the resource interfaces are the same between score-compose and score-k8s.
```suggestion
    connection: "mongodb://{{ .State.username }}:{{ .State.password }}@{{ .State.service }}:27017/"
```